lib/Runtime/Base/ThreadServiceWrapperBase.cpp

```diff
@@ -163,7 +163,22 @@ void ThreadServiceWrapperBase::FinishIdleCollect(ThreadServiceWrapperBase::Finis
     OnFinishIdleCollect();
 
     IDLE_COLLECT_TRACE(_u("Idle timer finished\n"));
-    JS_ETW(EventWriteJSCRIPT_GC_IDLE_FINISHED(this));
+
+    if (threadContext->GetRecycler()->CollectionInProgress() && !threadContext->IsInScript())
+    {
+        if (reason == FinishReason::FinishReasonTaskComplete)
+        {
+            // schedule another timer to check the progress
+            IDLE_COLLECT_TRACE(_u("FinishIdleCollect- collection is still in progress, schedule another timer to check the status\n"));
+            // schedule a shorter timer here to check the progress since the idle GC has already last more than one second and is likely to close to finish
+            // TODO: measure and adjust the timeout bellow
+            ScheduleIdleCollect(500, false);
```

Comment:
Put this in RecyclerHeuristic and allow this to be change via flag in debug/test build?

Reply:
will do